services/storage-users/pkg/command/uploads.go

```diff
@@ -311,3 +324,181 @@ func buildInfo(filter storage.UploadSessionFilter) string {
+		return nil
+	})
+
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error walking spaces root directory %s: %v\n", spacesRoot, err)
+		return nil
+	}
+
+	return spaceIDs
+}
+
+// delete stale processing nodes for a given spaceID
+func deleteStaleUploads(ctx *cli.Context, cfg *config.Config, spaceID string) {
+	fmt.Printf("\nDeleting stale processing nodes for space: %s\n", spaceID)
+
+	// Find .mpk files in space directory
+	spaceRoot := filepath.Join(cfg.Drivers.OCIS.Root, "spaces", lookup.Pathify(spaceID, 1, 2))
+	mpkFiles := []string{}
+	err := filepath.Walk(spaceRoot, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Error accessing path %s: %s\n", path, err)
+			return filepath.SkipDir
+		}
+		if !info.IsDir() && strings.HasSuffix(path, ".mpk") {
+			mpkFiles = append(mpkFiles, path)
+		}
+		return nil
+	})
+
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error walking space directory %s: %s\n", spaceRoot, err)
+		return
+	}
+
+	fmt.Printf("%sFound total %d .mpk files\n", LOG_INDENT_L1, len(mpkFiles))
+
+	staleCount := 0
+	for _, path := range mpkFiles {
+		staleCount += deleteStaleNode(ctx, cfg, path)
+	}
+
+	fmt.Printf("%sFound total %d stale nodes\n", LOG_INDENT_L1, staleCount)
+}
+
+// deleteStaleNode deletes a stale node if it is not referenced by any upload session
+// returns 1 if the node was deleted, 0 otherwise, for counting purposes
+func deleteStaleNode(ctx *cli.Context, cfg *config.Config, path string) int {
+	dryRun := ctx.Bool("dry-run")
+	verbose := ctx.Bool("verbose")
+	nodeDir := filepath.Dir(path)
+
+	// Read .mpk file to get processing info
+	b, err := os.ReadFile(path)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error reading file %s: %s\n", path, err)
+		return 0
+	}
+	var mpkData map[string]interface{}
+	if err := msgpack.Unmarshal(b, &mpkData); err != nil {
+		fmt.Fprintf(os.Stderr, "Error unmarshaling file %s: %s\n", path, err)
+		return 0
+	}
+
+	processingID := ""
+	hasProcessingNode := false
+	for k, v := range mpkData {
+		vStr := string(v.([]byte))
+		if k == MSGPACK_KEY_USER_OCIS_NODESTATUS && strings.Contains(vStr, node.ProcessingStatus) {
+			hasProcessingNode = true
+			processingID = strings.Split(vStr, ":")[1]
+		}
+	}
+	if !hasProcessingNode {
+		return 0
+	}
+
+	pathUploadInfo := filepath.Join(cfg.Drivers.OCIS.Root, "uploads", processingID) + ".info"
+	_, infoStatErr := os.Stat(pathUploadInfo)
+	infoExists := !os.IsNotExist(infoStatErr)
+	if infoExists {
+		return 0
+	}
+
+	if verbose {
+		fmt.Printf("%sFound stale upload at %s (Processing ID: %s)\n", LOG_INDENT_L1, path, processingID)
+		fmt.Printf("%sUpload info missing at: %s\n", LOG_INDENT_L2, pathUploadInfo)
+	}
+
+	if dryRun {
+		return 1
+	}
+
+	if err := os.RemoveAll(nodeDir); err != nil {
+		fmt.Fprintf(os.Stderr, "%sError deleting stale node %s: %v\n", LOG_INDENT_L2, nodeDir, err)
+	} else {
+		if verbose {
+			fmt.Printf("%sDeleted stale node: %s\n", LOG_INDENT_L2, nodeDir)
+		}
+		return 1
+	}
```

Comment:
Can we omit the if ... else logic? If we `return 0` in the `if` case we can just continue on main line.